shared/api/auth.go

```diff
@@ -24,3 +24,140 @@ const (
 	// IdentityTypeOIDCClient represents an identity that authenticates with OIDC..
 	IdentityTypeOIDCClient = "OIDC client"
 )
+
+// Identity is the type for an authenticated party that can make requests to the HTTPS API.
+//
+// swagger:model
+type Identity struct {
+	// AuthenticationMethod is the authentication method that the identity
+	// authenticates to LXD with.
+	// Example: tls
+	AuthenticationMethod string `json:"authentication_method" yaml:"authentication_method"`
+
+	// Type is the type of identity.
+	// Example: oidc-service-account
+	Type string `json:"type" yaml:"type"`
+
+	// Identifier is a unique identifier for the identity (e.g. certificate fingerprint or email for OIDC).
+	// Example: jane.doe@example.com
+	Identifier string `json:"id" yaml:"id"`
```

Comment:
If a TLS user updates their certificate by existing means does this also update their existing identity to reflect the new Identifier and Name?

Reply:
Yep that's correct.